Framework/Nexus/src/NexusClasses.cpp

```diff
@@ -221,14 +200,16 @@ bool NXClass::openLocal(const std::string &nxclass) {
 }
 
 void NXClass::close() {
-  if (NXclosegroup(m_fileID) == NXstatus::NX_ERROR) {
+  try {
+    m_fileID->closeGroup();
+  } catch (::NeXus::Exception const &) {
     throw std::runtime_error("Cannot close group " + name() + " of class " + NX_class() + " (trying to close path " +
                              m_path + ")");
   }
   m_open = false;
 }
 
-void NXClass::reset() { NXinitgroupdir(m_fileID); }
+void NXClass::reset() { m_fileID->initGroupDir(); }
```

Comment:
Comment (nothing to do for now), but this is a terrible name for this function. If I wasn't currently looking at the source I would have thought that it reset the filehande to the root group instead of resetting the iterator through the items in the current level.